conf/test_route_control.py

```diff
@@ -67,35 +65,43 @@ def test_given_valid_mac_when_mac_to_hex_then_return_hex_string_representation(
         self.assertEqual(mac_to_hex("00:1a:2b:3c:4d:5e"), "001A2B3C4D5E")
 
     def test_given_known_destination_when_fetch_mac_then_returns_mac(self):
-        ipdb = IPDB()
-        ipdb.nl.get_neighbours = lambda dst, **kwargs: [
-            {"attrs": [("NDA_DST", dst), ("NDA_LLADDR", "00:1a:2b:3c:4d:5e")]}
-        ]
-        self.assertEqual(fetch_mac(ipdb, "192.168.1.1"), "00:1a:2b:3c:4d:5e")
-
-    def test_given_unkonw_destination_when_fetch_mac_then_returns_none(self):
-        ipdb = IPDB()
-        ipdb.nl.get_neighbours = lambda dst, **kwargs: []
-        self.assertIsNone(fetch_mac(ipdb, "192.168.1.1"))
+        ndb = Mock()
+        kwargs = {
+            "ifindex": 1,
+            "dst": "192.168.1.1",
+            "lladdr": "00:1a:2b:3c:4d:5e"
+        }
+        ndb.neighbours.dump.return_value = [kwargs]
+        self.assertEqual(fetch_mac(ndb, "192.168.1.1"), "00:1a:2b:3c:4d:5e")
+
+    def test_given_unknown_destination_when_fetch_mac_then_returns_none(self):
+        ndb = Mock()
+        kwargs = {
+            "ifindex": 1,
+            "dst": "192.168.1.1",
+            "lladdr": None
+        }
+        ndb.neighbours.dump.return_value = [kwargs]
+        self.assertIsNone(fetch_mac(ndb, "192.168.1.1"))
 
 
 class TestRouteController(unittest.TestCase):
     def setUp(self):
         self.mock_bess_controller = Mock(BessControllerMock)
-        self.ipdb = Mock()
+        self.ndb = Mock()
         self.ipr = Mock()
         interfaces = ["access", "core"]
         self.route_controller = RouteController(
             self.mock_bess_controller,
-            self.ipdb,
+            self.ndb,
             interfaces=interfaces,
             ipr=self.ipr,
         )
 
     @patch("conf.route_control.fetch_mac")
-    @patch.object(RouteController, "get_merge_module_name")
-    @patch.object(RouteController, "get_route_module_name")
-    @patch.object(RouteController, "get_update_module_name")
+    @patch("conf.route_control.get_merge_module_name")
+    @patch("conf.route_control.get_route_module_name")
+    @patch("conf.route_control.get_update_module_name")
```

Comment:
Why these changes?

Reply:
In this PR, "get_merge_module_name", "get_route_module_name", "get_update_module_name" methods are converted to static methods as there is no necessity to define them as class methods.  That's the reason, we do not need to patch object any more. We started to  mock static methods.   If you think we do not need to mix it with this fix, I can remove this enhancement and  push it in another PR.

Reply:
It is ok. Thanks!